internal/model/requestquery.go

```diff
@@ -0,0 +1,55 @@
+package model
+
+import (
+	"errors"
+	"gorm.io/gorm"
+)
+
+type Status string
+type Type string
+
+const (
+	StatusOngoing   Status = "ongoing"
+	StatusAutoreply Status = "autoreply"
+	StatusPending   Status = "pending"
+	StatusClosed    Status = "closed"
+	StatusReviewed  Status = "reviewed"
+)
+
+const (
+	TypeUnknown Type = "unknown"
+	TypeQuery   Type = "query"
+	TypeRequest Type = "request"
+)
+
+type RequestQuery struct {
+	gorm.Model
+	Status    Status
+	Type      Type
+	BookingId *uint
+	Booking   *Booking
+	ChatId    uint
+	Chat      Chat
+	Messages  []Message
+}
+
+var ErrRequestHasNilBookingId = errors.New("booking id is required for requests")
+
+func (r *RequestQuery) Create(db *gorm.DB) error {
+	return db.Create(r).Error
+}
+
+func (r *RequestQuery) Update(db *gorm.DB) error {
+	return db.Updates(r).Error
+}
+
+func (r *RequestQuery) Delete(db *gorm.DB) error {
+	return db.Delete(r).Error
+}
+
+func (r *RequestQuery) BeforeSave(tx *gorm.DB) (err error) {
```

Comment:
Personally not a fan of named naked returns, but we can discuss about it

https://www.reddit.com/r/golang/comments/15rz5if/do_you_use_or_avoid_named_results/